src/parser/validator.ts

```diff
@@ -0,0 +1,52 @@
+import { RootParser } from "../parser/RootParser";
+import { makeCstVisitor, VisitorResult, CommandNode } from "../parser/makeCstVisitor";
+
+interface validateCSTResult {
+    errors: ValidateCSTError[];
+    hasErrors: boolean;
+}
+
+interface ValidateCSTError {
+    error: Error,
+}
+
+export function validateVisitorResult(visitorResult: VisitorResult): validateCSTResult {
+    var validateResult = { errors: [], hasErrors: false};
```

Comment:
Avoid `var` - prefer `const` unless you need to reassign the value, in which case use `let`. This should issue an eslint warning.